File: ios/TencentYlhAd.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(TencentYlhAd, NSObject)

RCT_EXTERN_METHOD(registerAppId:(NSString)appId
                  resolve:(RCTPromiseResolveBlock)resolve
                  reject:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(showInterstitialAD:(NSString)podId
                  asPopup:(BOOL)asPopup)

RCT_EXTERN_METHOD(showFullScreenAD:(NSString)podId)

RCT_EXTERN_METHOD(openWeb:(NSString)url
                  settings:(NSDictionary)settings)

@end

File: ios/TencentYlhAd.swift
@objc(TencentYlhAd)
class TencentYlhAd: NSObject {
    @objc(registerAppId:resolve:reject:)
    func registerAppId(appId: String, resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) -> Void {
        let result: Bool = GDTSDKConfig.registerAppId(appId);
        if result {
            resolve("")
        } else {
            reject("-1", "", nil)
        }
    }
    
    @objc(showInterstitialAD:asPopup:)
    func showInterstitialAD(podId: String, asPopup: Bool) -> Void {
        Interstitial.getInstance().showInterstitialAD(posID: podId, fullScreen: false);
    }
    
    @objc(showFullScreenAD:)
    func showFullScreenAD(podId: String) -> Void {
        Interstitial.getInstance().showInterstitialAD(posID: podId, fullScreen: true);
    }
    
    /***
       * h5
       * @param url
       * @param settings ["titleBarHeight", "titleBarColor", "title", "titleColor", "titleSize", "backButtonImage", "closeButtonImage", "separatorColor", "backSeparatorLength"]
       * titleColor='#ff0000ff',titleBarHeight=45 dp, titleSize=20 sp, backButtonImage='gdt_ic_back'
       */
    @objc(openWeb:settings:)
      func openWeb(url: String, settings: NSDictionary) {
        Hybrid.getInstance().openWeb(url: url, settings: settings)
      }
}
